spec/unit/puppet/parser/functions/echo_spec.rb

```diff
@@ -12,27 +12,27 @@
 
   context 'should output a correct debug string for' do
     it 'a string' do
-      expect(scope).to receive(:debug).with '(String) "test"'
+      allow(scope).to receive(:debug).with '(String) "test"'
```

Comment:
ok, I know I'm almost 5 years late, but was this change actually 'correct'?  Isn't the difference between `allow` and `expect` significant (and important here). (For context, someone was asking on slack how to test debug output from a puppet function.)

@ekohl ?

Reply:
You have `allow()` (=this can be called) and `expect()` (=this must be called). In this case the test looks like it should have used `expect()` since it must be called (that's pretty much the entire test).

The alternatives are [spies](https://relishapp.com/rspec/rspec-mocks/docs/basics/spies) where you can use `allow()` to set it up, then act (`run.with_params()`) and verify `expect(scope).to have_received(:debug)`.